src/category_theory/monad/bundled.lean

```diff
@@ -0,0 +1,144 @@
+/-
+Copyright (c) 2020 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+import category_theory.monad.basic
+import category_theory.eq_to_hom
+
+/-!
+# Bundled Monads
+
+We define bundled (co)monads as a structure consisting of a functor `func : C ⥤ C` endowed with
+a term of type `(co)monad func`. See `category_theory.monad.basic` for the definition.
+The type of bundled (co)monads on a category `C` is denoted `(Co)Monad C`.
+
+We also define morphisms of bundled (co)monads as morphisms of their underlying (co)monads
+in the sense of `category_theory.(co)monad_hom`. We construct a category instance on `(Co)Monad C`.
+-/
+
+namespace category_theory
+open category
+
+universes v u -- declare the `v`'s first; see `category_theory.category` for an explanation
+
+variables (C : Type u) [category.{v} C]
+
+/-- Bundled monads. -/
+structure Monad :=
+(func : C ⥤ C)
+(str : monad func . tactic.apply_instance)
+
+/-- Bundled comonads -/
+structure Comonad :=
+(func : C ⥤ C)
+(str : comonad func . tactic.apply_instance)
+
+namespace Monad
+
+/-- The initial monad. TODO: Prove it's initial. -/
+def initial : Monad C := { func := 𝟭 _ }
+
+variable {C}
+
+instance : inhabited (Monad C) := ⟨initial C⟩
+
+instance {M : Monad C} : monad M.func := M.str
+
+/-- Morphisms of bundled monads. -/
+def hom (M N : Monad C) := monad_hom M.func N.func
+
+namespace hom
+instance {M : Monad C} : inhabited (hom M M) := ⟨monad_hom.ident _⟩
+end hom
+
+instance : category (Monad C) :=
+{ hom := hom,
+  id := λ _, monad_hom.ident _,
+  comp := λ _ _ _, monad_hom.comp,
+  id_comp' := λ _ _, by apply monad_hom.ident_comp,
+  comp_id' := λ _ _, by apply monad_hom.comp_ident,
+  assoc' := λ _ _ _ _, by apply monad_hom.comp_assoc }
```

Comment:
```suggestion
  id_comp' := λ _ _, monad_hom.ident_comp,
  comp_id' := λ _ _, monad_hom.comp_ident,
  assoc' := λ _ _ _ _, monad_hom.comp_assoc }
```

Reply:
Or just leave out these fields entirely!

Reply:
Done. I'm just relying on the auto-param now.